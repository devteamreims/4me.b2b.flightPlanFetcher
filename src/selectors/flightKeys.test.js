import R from 'ramda';

import {
  getKeysFromIfplId,
  getKeysFromCallsign,
} from './flightKeys';

const testState = {
  flightKeys: {
    'AA123': {
      callsign: 'AFR1234',
    },
    'AA456': {
      callsign: 'AFR4321',
    },
  },
};

describe('flightKeys selectors', () => {
  describe('getByIfplId', () => {
    test('should return the keys for an existing ifplId', () => {
      expect(getKeysFromIfplId('AA123')(testState)).toBeDefined();
    });

    test('should return null for a non existing ifplId', () => {
      expect(getKeysFromIfplId('non-existent')(testState)).toBe(null);
    });

    test('should include ifplId in returned object', () => {
      const keys = getKeysFromIfplId('AA123')(testState);
      expect(keys.ifplId).toBe('AA123');
    });
  });

  describe('getByCallsign', () => {
    test('should return the keys for an existing callsign', () => {
      expect(getKeysFromCallsign('AFR1234')(testState)).toMatchSnapshot();
    });
  });
});
