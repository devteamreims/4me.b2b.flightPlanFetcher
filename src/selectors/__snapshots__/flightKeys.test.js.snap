exports[`flightKeys selectors getByCallsign should return the keys for an existing callsign 1`] = `
Object {
  "callsign": "AFR1234",
  "ifplId": "AA123",
}
`;
